src/systems/tracked-controls.js

```diff
@@ -38,11 +36,23 @@ module.exports.System = registerSystem('tracked-controls', {
     var gamepads;
     var i;
     var prevCount;
+    var thisCount;
 
     gamepads = navigator.getGamepads && navigator.getGamepads();
     if (!gamepads) { return; }
 
+    // Count first, and only actually update controller list if length changes.
     prevCount = controllers.length;
+    thisCount = 0;
```

Comment:
OK.  I do think that if we can find some way to detect that there are no controller identity changes and not emit the controllersupdated event, that would be a significant savings.